Page updates: column-rule-* (#44755)
mdn/content: files/en-us/web/css/reference/properties/column-rule-style/index.md

sidebar: cssref
---

> The **`column-rule-style`** [CSS](/en-US/docs/Web/CSS) property defines the line style of the lines drawn between columns in multi-column grid, flex, and multi-col layouts.

Reviewer: I "fixed" other example of multi-col layouts to multi-column layouts. If this is reasonable, also needs to be fixed everywhere.

Author: it think it needs to be multi-col. it was multi-col, and it reads "multi-column grid" in the same sentence, so needs to be differentiated

> {{InteractiveExample("CSS Demo: column-rule-style")}}

Reviewer: Doesnt work at all because only one column is displayed.

Author: fixed

```css interactive-example

> #example-element {

Reviewer: Line duplicated below

```suggestion
```

Author: this is what made the example fail.

column-rule-style: unset;
```

### Values

The `column-rule-style` property accepts a comma-separated list of values, including:

- `<line-style>`
  - : A {{cssxref("&lt;line-style&gt;")}}, either `none`, `hidden`, `dotted`, `dashed`, `solid`, `double`, `groove`, `ridge`, `inset`, or `outset`. The default value is`none`.

- `<repeat-line-style>`

>   - : A {{cssxref("repeat()")}} function, with the first parameter being an {{cssxref("&lt;integer&gt;")}} of `1` or more, representing the number of times the second parameter, a comma-separated list of {{cssxref("&lt;line-style&gt;")}} values, should be repeated.

Reviewer: @estelle In general, since there is a lot of copy-pasting going on, can you also capture the fixed from @chrisdavidmills in https://github.com/mdn/content/pull/44725

For example, based on https://github.com/mdn/content/pull/44725/changes#r3595366823 we'd do:

```suggestion
  - : A {{cssxref("repeat()")}} function, with the first argument being an {{cssxref("&lt;integer&gt;")}} of `1` or more, and one or more {{cssxref("&lt;line-width&gt;")}} values as subsequent arguments. The integer defines the number of times the `<line-width>` values should be repeated.
```

- `<auto-repeat-line-style>`
  - : A {{cssxref("repeat()")}} function, with `auto` as the first argument and one or more `<line-style>` values as subsequent arguments. The provided `<line-style>` values are repeated as many times as needed to fill in values for any column-rules that are not explicitly specified by other components of the property value.

## Description

The `column-rule-style` property defines the line style of any column rule lines drawn in the gaps between columns in [multi-column](/en-US/docs/Web/CSS/Guides/Multicol_layout), [flex](/en-US/docs/Web/CSS/Guides/Flexible_box_layout), and [grid](/en-US/docs/Web/CSS/Guides/Grid_layout) containers with more than one column.

> The value is a comma-separated list of components, which can include `<line-style>`, `<repeated-line-style>`, and `<auto-repeat-line-style>` types.

Reviewer: ```suggestion
The value is a comma-separated list of components, which can include `<line-style>`, `<repeat-line-style>`, and `<auto-repeat-line-style>` types.
```

The `column-rule-style`, along with the {{cssxref("column-rule-color")}} and {{cssxref("column-rule-width")}} properties, can be set using the {{cssxref("column-rule")}} shorthand. The `column-rule-style`, along with the {{cssxref("row-rule-style")}} property, can also be set using the {{cssxref("rule-style")}} shorthand.

If the property value has only one `<line-style>`, all the column rules will be that style. If we declare the following, all column rules will be `double`:

```css

> column-rule-style: `double`;

Reviewer: ```suggestion
column-rule-style: double;
```

column-rule-style: solid, outset, inset, outset, inset;
column-rule-style: solid, repeat(2, outset, inset);
```

This creates a list of five styles. If the number of styles in the `column-rule-style` value's style list exceeds the number of gaps between columns, the excess style values are ignored. If the container has three columns, the rule in the first gutter will be `solid` and the second `outset`.

If there are more gutters than styles, as there are no `repeat()` functions that include the keyword `auto`, the list of styles is repeated. If the container has 6, 11, 16 or 21 columns, this sequence of styles will be repeated one, two, three, or four times, respectively, with the last rule being `inset`.

### Auto-repeating line styles

The `repeat()` function also accepts `auto` as the first argument instead of a positive integer. With `auto` as the first argument, the comma-separated list of `<line-style>` values passed as the second parameter will be repeated as many times as needed to fill in values for any column-rules that are not explicitly specified by other components of the property value, if any.

```css
column-rule-style: solid, repeat(auto, dotted), solid;
```

> In this case, the first and last column rules will be `sold`, and all others will be `dotted`. It doesn't matter if the container has 3, 6, 11, 16, or 21 columns, the first and last columns will always be `thin`. All the other column rules will be `dotted`, which means if there are only 2 or 3 columns, there will be no dotted column rules.

Reviewer: Typo 
```suggestion
In this case, the first and last column rules will be `solid`, and all others will be `dotted`. It doesn't matter if the container has 3, 6, 11, 16, or 21 columns, the first and last columns will always be `thin`. All the other column rules will be `dotted`, which means if there are only 2 or 3 columns, there will be no dotted column rules.
```

The `column-rule-style` property accepts a comma-separated list of values, including:

- `<line-style>`
  - : A {{cssxref("&lt;line-style&gt;")}}: one of `none`, `hidden`, `dotted`, `dashed`, `solid`, `double`, `groove`, `ridge`, `inset`, or `outset`. The default value is `none`.

- `<repeat-line-style>`

>   - : A {{cssxref("repeat()")}} function, with the first argument being an {{cssxref("&lt;integer&gt;")}} of `1` or more, and subsequent arguments being {{cssxref("&lt;line-style&gt;")}} values. The integer specifies how many times the `<line-style>` values should be repeated.

Reviewer: Typo!

#### Result

{{ EmbedLiveSample('basic usage') }}

### Multiple values

#### HTML

We include a list of authors:

```html live-sample___multiple live-sample___repeat live-sample___func live-sample___auto
<ul>
  <li>Kimberlé Crenshaw</li>
  <li>Angela Y. Davis</li>
  <li>Bernardine Evaristo</li>
  <li>Audre Lorde</li>
  <li>Cathy Park Hong</li>
  <li>Zoya Patel</li>
  <li>Juno Mac</li>
  <li>Molly Smith</li>
  <li>Tara Westover</li>
</ul>
```

#### CSS

We define the list to be a flex container, creating columns by setting the {{cssxref("flex-direction")}} to `row` using the {{cssxref("flex-flow")}} shorthand. We include a {{cssxref("gap")}} of `5px` to provide enough room between the columns to fit our `3px` double teal rule:

```css live-sample___multiple live-sample___repeat live-sample___func live-sample___auto
ul {
  display: flex;
  flex-flow: row;
  gap: 5px;
  column-rule-width: 3px;
  column-rule-color: teal;

  column-rule-style:
    dotted, dashed, solid, double, groove, ridge, inset, outset, none, hidden;
}
```

#### Result

> {{EmbedLiveSample("Multiple", "", "180")}}

Reviewer: FWIW the use of multiple here seems to work. The https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Live_samples doesn't state whether the identifier is case sensitive or not.

Author: They are not case sensitive. EmbedLiveSample first parameter is very forgiving actually

As there are more values (10) than gutters (8), the `none` and `hidden` values are not used.

### Repeating values

This example demonstrates how, when there are fewer values in the list of styles than column rules, the values are repeated.

Using the same HTML and CSS as in the previous example, we include three comma-separated styles as the `column-rule-style` value:

```css live-sample___repeat
ul {
  column-rule-style: solid, groove, double;
}
```

{{EmbedLiveSample("Repeat", "", "180")}}

### Using the `repeat()` function

> This example demonstrates using the `repeat()` function within the `column-rule-style` property value. We use the same HTML and CSS as in the previous examples. We include a `repeat()` function, setting the list of two `<line-style>` values to be repeat 3 times.

Reviewer: ```suggestion
This example demonstrates using the `repeat()` function within the `column-rule-style` property value. We use the same HTML and CSS as in the previous examples. We include a `repeat()` function, setting the list of two `<line-style>` values to be repeated 3 times.
```

```css live-sample___func live-sample___auto
ul {
  --base: 0.5vw;

>   --secondary: 1vw;

Reviewer: Claude says these are unused elsewhre in the example (copy paste)
